src/scan.l

```diff
@@ -285,9 +285,24 @@ M4QEND      "]""]"
 }
 
 <CODEBLOCK_MATCH_BRACE>{
+    "'}'"      |
+    "'{'"      |
+    [^{}\r\n]  |
+    \"(\\\"|[^"\r\n])*\"  |
```

Comment:
I think in C, backslash in end-of-line to make multi-line string literal is allowed.
```
"something like\
this"
```

Reply:
Correct, but fairly seldomly used. flex will report an error to user in this case who could then simply adjust the code. We should not overcomplicate things here and target a perfect parser given that no user filed a complaint about that. I am opting to not tackle all edgy corners of the c/c++ syntax.

But we are happy to review any enhancement.

Reply:
I would argue that all of your cases you made in this PR can all be worked-around. That why I ask you for real world code that has been broken by the "bug" you noted in the title.

Reply:
It is clear that this is a bug, since according to the manual top code blocks are same as `%{ ... %}` code blocks. For a real life example put code documentation into the top section which somewhere contains a `}`.

Not sure which workarounds you are thinking of, but if valid code cannot be put into the top code section (even without any far-reaching syntax exceptions) causing the user to fiddle around is still a bug.